### PR TITLE
[PAY-1342][PAY-1360] DMs: Show users as chat unlocked by default, optimistic chat navigation

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Defaults users to chatable before they're loaded in
- Navigate optimistically to chats from compose modal

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-default-true

File: packages/common/src/store/pages/chat/sagas.ts [hunk 1]
```diff
@@ -214,6 +214,11 @@ function* doCreateChat(action: ReturnType<typeof createChat>) {
       .map((id) => encodeHashId(id))
       .sort()
       .join(':')
+
+    // Optimistically navigate - if we fail we'll toast
+    yield* put(goToChat({ chatId }))
```
Comment (dharit-tan) [inline packages/common/src/store/pages/chat/sagas.ts:219]: Does this fail often on a newly created chat? I know we discussed this but wondering what you observed.

File: packages/common/src/store/pages/chat/slice.ts [hunk 2]
```diff
@@ -131,7 +131,10 @@ const slice = createSlice({
     },
     createChatSucceeded: (state, action: PayloadAction<{ chat: UserChat }>) => {
       const { chat } = action.payload
-      chatsAdapter.upsertOne(state.chats, chat)
+      chatsAdapter.upsertOne(state.chats, {
+        ...chat,
+        messagesStatus: Status.IDLE // ensures ChatScreen/ChatMessagesList show "Say Hi"
```
Comment (dharit-tan) [inline packages/common/src/store/pages/chat/slice.ts:136]: nice

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx [hunk 3]
```diff
@@ -244,8 +244,8 @@ export const ChatScreen = () => {
   const popupMessage = useSelector((state) =>
     getChatMessageById(state, chatId ?? '', popupMessageId ?? '')
   )
-  const { canSendMessage, firstOtherUser: otherUser } =
-    useCanSendMessage(chatId)
+  const { canSendMessage, firstOtherUser } = useCanSendMessage(chatId)
+  const otherUser = firstOtherUser ?? {}
```
Comment (dharit-tan) [inline packages/mobile/src/screens/chat-screen/ChatScreen.tsx:248]: Is this better than leaving undefined? I think later in the file I am accessing `otherUser.user_id` conditionally on `!!otherUser`.

**Review by piazzatron — APPROVED** [review]
Wow, this change makes a HUGE difference in the perceived snappiness of the product. Great stuff!!

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx (hunk 3 above)
Comment (rickyrombo) [reply packages/mobile/src/screens/chat-screen/ChatScreen.tsx:248]: interesting, if `otherUser` is undefined the component fails to render due to https://github.com/AudiusProject/audius-client/pull/3514/files#diff-599f9b15aa87cbaec942aaf03242204cf8cd103687bde3f1416bc0e039af3a68R287

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-default-true

**Pull request merged**